relay/sources/rudder-pkg/lib/rudder-pkg/rpkg.py

```diff
@@ -0,0 +1,151 @@
+            utils.fail("The version %s does not respect the version syntax"%(pluginLongVersion))
+
+        if match.group('rudderVersion') is None or match.group('pluginShortVersion') is None:
+            utils.fail("The version %s does not respect the version syntax [0-9]+.[0-9]+-[0-9]+.[0-9]+(-SNAPSHOT){0,1}"%(pluginLongVersion))
+        else:
+            self.rudderVersion = match.group('rudderVersion')
+            self.pluginShortVersion = match.group('pluginShortVersion')
+            self.pluginLongVersion = pluginLongVersion
+
+    def __str__(self):
+        return self.mode + " => " + self.rudderVersion + "-" + self.pluginShortVersion + " | " + self.pluginLongVersion
+
+    def __repr__(self):
+        return self.mode + " => " + self.rudderVersion + "-" + self.pluginShortVersion + " | " + self.pluginLongVersion + "\n"
+
+    def __hash__(self):
+        return hash((self.mode, self.rudderVersion, self.pluginShortVersion, self.pluginLongVersion))
+
+    def __eq__(self, other):
+        if isinstance(other, Version):
+            return self.mode == other.mode and self.rudderVersion == other.rudderVersion and self.pluginShortVersion == other.pluginShortVersion and self.pluginLongVersion == other.pluginLongVersion
+        return False
+
+    def __lt__(self, other):
+        return parse_version(self.pluginLongVersion) < parse_version(other.pluginLongVersion)
+        
+    def __le__(self, other):
+        return parse_version(self.pluginLongVersion) <= parse_version(other.pluginLongVersion)
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def __gt__(self, other):
+        return parse_version(self.pluginLongVersion) > parse_version(other.pluginLongVersion)
+
+    def __ge__(self, other):
+        return parse_version(self.pluginLongVersion) >= parse_version(other.pluginLongVersion)
+
+"""
+    Define an object based on a .rpkg file.
+"""
+class Rpkg:
+    def __init__(self, longName, shortName, path, version, metadata): 
+        self.longName = longName
+        self.shortName = shortName
+        self.path = path
+        self.version = version
+        self.metadata = metadata
+
+    def verifyAuth(self):
+        # Get rudder plugins gpg key
+        utils.getRudderKey()
+        # Download hash and sign files
+        (folder, leaf) = os.path.split(self.path)
+
+        shaSumUrl = folder + "/" + "SHA512SUMS"
+        shaSumDst   = utils.FOLDER_PATH + "/" + shaSumUrl
+        logging.info("downloading shasum file  %s"%(utils.URL + "/" + shaSumUrl))
+        utils.download(utils.URL + "/" + shaSumUrl)
+
+        signUrl   = folder + "/" + "SHA512SUMS.asc"
+        signDst   = utils.FOLDER_PATH + "/" + signUrl
+        logging.info("downloading shasum sign file  %s"%(utils.URL + "/" + signUrl))
+        utils.download(utils.URL + "/" + signUrl)
+
+        # Verify that signature is correct
+        gpgCommand = "/usr/bin/gpg --homedir " + utils.GPG_HOME + " --verify " + signDst + " " + shaSumDst
```

Comment:
Shouldn't we put trust data (gpg homedir) withing rudder installation and not the user's ?